src/Runner.Common/ResultsServer.cs

```diff
@@ -94,5 +122,128 @@ public void InitializeResultsClient(Uri uri, string token)
+        {
+            try
+            {
+                Trace.Info($"Attempting to start websocket client with delay {delay}.");
+                await Task.Delay(delay);
+                using var connectTimeoutTokenSource = new CancellationTokenSource(TimeSpan.FromSeconds(30));
+                await this._websocketClient.ConnectAsync(new Uri(feedStreamUrl), connectTimeoutTokenSource.Token);
+                Trace.Info($"Successfully started websocket client.");
+            }
+            catch (Exception ex)
+            {
+                Trace.Info("Exception caught during websocket client connect, fallback of HTTP would be used now instead of websocket.");
+                Trace.Error(ex);
+                this._websocketClient = null;
+            }
+        }
+
+        public async Task AppendLiveConsoleFeedAsync(Guid scopeIdentifier, string hubName, Guid planId, Guid timelineId, Guid timelineRecordId, Guid stepId, IList<string> lines, long? startLine, CancellationToken cancellationToken)
+        {
+            if (_websocketConnectTask != null)
+            {
+                await _websocketConnectTask;
+            }
+
+            // "_websocketClient != null" implies either: We have a successful connection OR we have to attempt sending again and then reconnect
+            // ...in other words, if websocket client is null, we will skip sending to websocket and just use rest api calls to send data
+            if (_websocketClient != null)
+            {
+                var linesWrapper = startLine.HasValue ? new TimelineRecordFeedLinesWrapper(stepId, lines, startLine.Value) : new TimelineRecordFeedLinesWrapper(stepId, lines);
+                var jsonData = StringUtil.ConvertToJson(linesWrapper);
+                try
+                {
+                    _totalBatchedLinesAttemptedByWebsocket++;
+                    var jsonDataBytes = Encoding.UTF8.GetBytes(jsonData);
+                    // break the message into chunks of 1024 bytes
+                    for (var i = 0; i < jsonDataBytes.Length; i += 1 * 1024)
+                    {
+                        var lastChunk = i + (1 * 1024) >= jsonDataBytes.Length;
+                        var chunk = new ArraySegment<byte>(jsonDataBytes, i, Math.Min(1 * 1024, jsonDataBytes.Length - i));
+                        await _websocketClient.SendAsync(chunk, WebSocketMessageType.Text, endOfMessage: lastChunk, cancellationToken);
+                    }
+
+                }
+                catch (Exception ex)
+                {
+                    _failedAttemptsToPostBatchedLinesByWebsocket++;
+                    Trace.Info($"Caught exception during append web console line to websocket, let's fallback to sending via non-websocket call (total calls: {_totalBatchedLinesAttemptedByWebsocket}, failed calls: {_failedAttemptsToPostBatchedLinesByWebsocket}, websocket state: {this._websocketClient?.State}).");
```

Comment:
That's a really good question...  hmmm,  I feel we could retry connection for couple more times, but if this still doesn't work, there won't be any fallback.

Reply:
can we check telemetry and see if the current failback to REST for the jobserver ever brings any value? ex: how many jobs actually need to fallback to REST API, maybe the answer is we don't need any fallback if the websocket is stable.

Reply:
We aren't planning for any fallback in Results at the moment.  I do see quite a few calls to this REST API today, but I also suspect that's because we don't retry deliver console log via websocket connection, and fallback with the first error.  I am going to add retries here first.

Reply:
> What should be the correct behavior for websocket exceptions when using the result service?

Also this codepath is exercised only with the new server logic.  We would still use `jobServer` with current server logic, even if live log through Results (the `feedStreamUrl` points to new results).